LanguageFeatures/Records/record_type_annotations_A02_t06.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The grammar is:
+///
+/// // Existing rules:
+/// type                   ::= functionType '?'?      // Existing production.
+///                          | recordType             // New production.
+///                          | typeNotFunction        // Existing production.
+///
+/// typeNotFunction        ::= 'void'                 // Existing production.
+///                          | recordType             // New production.
+///                          | typeNotVoidNotFunction // Existing production.
+///
+/// // New rules:
+/// recordType            ::= '(' recordTypeFields ',' recordTypeNamedFields ')'
+///                          | '(' recordTypeFields ','? ')'
+///                          | '(' recordTypeNamedFields ')'
+///
+/// recordTypeFields       ::= recordTypeField ( ',' recordTypeField )*
+/// recordTypeField        ::= metadata type identifier?
+///
+/// recordTypeNamedFields  ::= '{' recordTypeNamedField
+///                            ( ',' recordTypeNamedField )* ','? '}'
+/// recordTypeNamedField   ::= type identifier
+/// recordTypeNamedField   ::= metadata typedIdentifier
+///
+/// The grammar is exactly the same as parameterTypeList in function types but
+/// without `()`, `required`, and optional positional parameters since those
+/// don't apply to record types. A record type can't appear in an extends,
+/// implements, with, or mixin on clause, which is enforced by being a
+/// production in `type` and not `typeNotVoid`.
+///
+/// @description Checks that it is a compile-time error if record type
+/// appear in an `on` clause
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=records
+
+typedef Rec = (int i, String s);
+
+mixin M1 on (int i, String s) {}
+//          ^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+mixin M2 on Rec {}
+//          ^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+class MA1 with M1 {}
+class MA2 with M2 {}
+
+main() {
+  MA1();
+  MA2();
+}
```

Comment:
But an extension `on` a record type should be fine. Do we have that already?

Reply:
We didn't but now we do
https://github.com/sgrekhov/co19/blob/co19-1393/LanguageFeatures/Records/record_type_annotations_A01_t05.dart
https://github.com/sgrekhov/co19/blob/co19-1393/LanguageFeatures/Records/record_type_annotations_A01_t06.dart

Reply:
Thx!